providers/instantly/write.go

```diff
@@ -0,0 +1,121 @@
+package instantly
+
+import (
+	"context"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/handy"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+	"github.com/spyzhov/ajson"
+)
+
+// Write method allows to
+// * create campaigns
+// * create/update email-accounts
+// * create client
+// Documentation links can be found within constructURLPath function.
+func (c *Connector) Write(
+	ctx context.Context, config common.WriteParams,
+) (*common.WriteResult, error) {
+	if len(config.ObjectName) == 0 {
+		return nil, common.ErrMissingObjects
+	}
+
+	if !supportedObjectsByWrite.Has(config.ObjectName) {
+		return nil, common.ErrOperationNotSupportedForObject
+	}
+
+	// Object name will be used to complete further URL construction below.
+	url, err := c.getURL()
+	if err != nil {
+		return nil, err
+	}
+
+	var write common.WriteMethod
+	if len(config.RecordId) == 0 {
+		write = c.Client.Post
+		constructURLPathCreate(config, url) // nolint:wsl
+	} else {
+		write = c.Client.Patch
+		constructURLPathUpdate(config, url)
+	}
+
+	res, err := write(ctx, url.String(), config.RecordData)
+	if err != nil {
+		return nil, err
+	}
+
+	if res == nil || res.Body == nil {
+		// it is unlikely to have no payload
+		return &common.WriteResult{
+			Success: true,
+		}, nil
+	}
+
+	recordIdNodePath := recordIdPaths[config.ObjectName]
+
+	// write response was with payload
+	return constructWriteResult(res.Body, recordIdNodePath)
+}
+
+var recordIdPaths = map[string]*string{ // nolint:gochecknoglobals
+	objectNameLeads:            nil, // ID is not returned for Leads.
```

Comment:
however, emails are used to identify a lead along with campaign id. we should handle this.

Reply:
I will close this comment as we are discussing this in the thread above.